### PR TITLE
DM-32940: Use butler.exists instead of datasetExists

### PR DESCRIPTION
**Pull request by timj**
Requires lsst/daf_butler#840

### CODE REVIEW

File: tests/test_butler.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def testExecutionExistence(self):
             with self.subTest(chain=chain):
                 datasets = self._get_datasets_from_chain(chain)
                 for ref in datasets:
-                    self.assertTrue(self.butler.datasetExists(ref, collections=chain))
+                    self.assertTrue(self.butler.exists(ref))
```
Comment (timj) [inline tests/test_butler.py:114]: This is one case where `exists_many()` would have been useful but it's not critical given the number of refs being tested.

Comment (timj) [reply tests/test_butler.py:114]: This is nominally a middleware package so in theory there is no downside to using `_exists_many()` here and fixing the code when we make the interface formally public.

Comment (andy-slac) [reply tests/test_butler.py:114]: I have no strong opinion, either way should work OK.

**Review by andy-slac — APPROVED** [review]
Looks good.

**Pull request merged**